### ISSUE TITLE
Consider using pollyjs

### ISSUE THREAD

**Issue by mosch** [allthings/node-sdk#146]
Right now, we are doing http requests in testing. This can be slow and unreliable. Multiple different approaches have been discussed before, maybe [this is a new one](https://github.com/Netflix/pollyjs) we could look at.

/discuss @svlapin @iamn1ck @adieuadieu

**Comment by adieuadieu** [comment]
Oh look! Netflix had the same problem we did! 🎉 It sounds like it could be what we are looking for.

**Comment by svlapin** [comment]
Oh that might be a nice idea. I remember myself using [a similar one](https://github.com/vcr/vcr) in Ruby project previously.

I have one concern though - if we expose the recordings in this public repo, we expose a snapshot of complete communication with our API server, including tokens and other internals. Is that ok from a security standpoint?

**Comment by mosch** [comment]
> I have one concern though - if we expose the recordings in this public repo, we expose a snapshot of complete communication with our API server, including tokens and other internals. Is that ok from a security standpoint?

I don't think there's anything we need to hide. I guess it will be the tokens from our dev setup only?

**Comment by adieuadieu** [comment]
Yes. We'd just need to scrub or mock the token (even though they expire after 60 min) and the OAuth Client details that get stored in the snapshot.
